qiskit_nature/second_q/operators/second_quantized_op.py

```diff
@@ -71,7 +71,22 @@ def is_hermitian(self, atol: Optional[float] = None) -> bool:
         Returns:
             True if the operator is hermitian up to numerical tolerance, False otherwise.
         """
+        return self.equiv(self.adjoint(), atol=atol)
+
+    def equiv(self, other: Any, atol: Optional[float] = None) -> bool:
+        """Checks whether this operator is approximately equal to another operator.
+
+        Args:
+            other: The operator to compare to for approximate equality.
+            atol: Absolute numerical tolerance. The default behavior is to use ``self.atol``,
+                which would be 1e-8 unless changed by the user.
+
+        Returns:
+            True if the operators are equal up to numerical tolerance, False otherwise.
+        """
+        if not isinstance(other, type(self)):
+            return NotImplemented
```

Comment:
This should `raise NotImplementedError` and provide both, the type of `self` and of `other`.

Reply:
Done.

Reply:
Actually I realized it should be TypeError (see https://docs.python.org/3/library/exceptions.html) so I changed it.